README.md

```diff
@@ -96,6 +96,53 @@ status:
 ...
 ```
 
+## End to end Testing
+
+There is two type of e2e tests launched on CI.
+
+The first one would just apply the yaml files making sure they don't have any
+syntax issues. Pretty simple one, it just basically check the syntax.
+
+The second one would do some proper functional testing, making sure the task
+actually **ran** properly.
+
+The way the functional tests works is that if you have a directory called
```

Comment:
I thought a bit about it but I think unfortunately that some tasks are going to be pretty hard to test properly. Some of them may depend to an external service that's going to be hard to setup. 

Takes for example `argocd` and `openwhisk` task, would we need to pre-setup an argocd or openwhisk cluster on every PR ?

or If we start having some proposal of tasks that needs some sort of external API access are we going to ask the user to be able to provide the tokens and access so we can do the testing or are we going to reject this kind of tasks?

Reply:
We need to add tests on the existing one, also making it so that it's easy to write those tests and thus being able to say "except in some really exceptionnal cases, tests should be there)

Reply:
@vdemeester that doesn't answer my questions 😁 , are you suggesting we are enforcing  to spin up an openwhisk, knative or argocd cluster on every PR ? or not allowing tasks that have external api accesses?

Reply:
@chmouel not yet :wink: not really or "not yet". We can insensitive user to write test if it's simple to write them. We may enforce them later on :angel:. Also we *should* only run the tests related to the files that changed or have been added — and have a nightly that runs *all*.

Reply:
@chmouel yep, we'll see what cost we are ready to take :stuck_out_tongue:

Reply:
whatever get decided, I suggest this is enforced/added in a future iteration....

Reply:
Yeah, that's what I am telling : "We may enforce them later on :angel:"